### PR TITLE
Add status matchers to help unify test code.

### PR DESCRIPTION
**Pull request by markusthoemmes**
Some of our tests use a common pattern to match these, but some fail to provide all the debug information we want. This adds generalized versions to be used in tests so test output is uniform.

### CODE REVIEW

File: test/request.go [hunk 1]
```diff
@@ -50,6 +50,22 @@ func Retrying(rc spoof.ResponseChecker, codes ...int) spoof.ResponseChecker {
 	}
 }
 
+// IsStatusCode checks that the response code is equal to the given one.
+func IsStatusCode(code int) spoof.ResponseChecker {
```
Comment (vagababov) [inline test/request.go]: Would it be useful, to extend this to `codes []int` and check if any of matches? E.g. below we can check for 200 and 201.

Comment (markusthoemmes) [reply test/request.go]: In this case, yes.

I won't change the Matcher below though. I'd like us to write as tight checks as possible (I'm in the process of cleaning them all up). If we expect a 201, we should explicitly wait for that in the test.

File: test/request.go [hunk 2]
```diff
@@ -50,6 +50,24 @@ func Retrying(rc spoof.ResponseChecker, codes ...int) spoof.ResponseChecker {
...
+// IsOneOfStatusCodes checks that the response code is equal to the given one.
+func IsOneOfStatusCodes(codes ...int) spoof.ResponseChecker {
+	return func(resp *spoof.Response) (bool, error) {
+		for code := range codes {
+			if resp.StatusCode == code {
+				return true, nil
+			}
+		}
+
+		return true, fmt.Errorf("status = %d, want one of: %v, body = %s", resp.StatusCode, codes, string(resp.Body))
```
Comment (srinivashegde86) [inline test/request.go]: return false, err?

Comment (markusthoemmes) [reply test/request.go]: @srinivashegde86 this is consistent with the other matchers (see `MatchesBody` below).

Comment (srinivashegde86) [reply test/request.go]: oops. this bool represents whether this is done and not indicate if it passes or fails..

**Pull request merged**